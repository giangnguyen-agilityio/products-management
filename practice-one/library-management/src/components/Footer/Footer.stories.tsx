import { type StoryFn, type Meta } from "@storybook/react";
import Footer from "./index";

export default {
  title: "Example/Footer",
  component: Footer,
} as Meta<typeof Footer>;

const Template: StoryFn<typeof Footer> = () => <Footer />;
export const Default = Template.bind({});
