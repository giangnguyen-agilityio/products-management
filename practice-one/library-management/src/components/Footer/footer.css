.site-footer {
  display: flex;
  width: 100%;
  justify-content: space-between;
  margin: 40px 0 20px;
  border-top: 2px solid var(--primary-color);
}

.copyright {
  padding: 0 0 0 25px;
  font-size: 18px;
  letter-spacing: 0.8px;
  line-height: 1.5;
  color: var(--secondary-color);
}

.link-social {
  display: flex;
  gap: 20px;
  align-items: center;
  padding: 15px 0 0 25px;
}

.link-social-item {
  color: var(--secondary-color);
}

.link-social-item:hover {
  color: var(--primary-color-text);
}

.footer-right {
  display: flex;
  gap: 100px;
  padding: 40px 0 0;
}

.footer-column {
  display: flex;
  flex-direction: column;
  gap: 20px;
}

.footer-column-title {
  font-family: var(--primary-font);
  font-size: 28px;
  text-transform: capitalize;
}

.footer-column-items {
  display: flex;
  flex-direction: column;
  gap: 15px 0;
}

.footer-column-link {
  color: var(--secondary-color);
  transition: all 0.3s ease-in-out;
}

.footer-column-link:hover {
  color: var(--primary-color-text);
}
