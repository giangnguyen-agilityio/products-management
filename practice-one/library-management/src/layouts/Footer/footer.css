.site-footer {
  display: flex;
  width: 100%;
  justify-content: space-between;
  margin: 40px 0 0;
  border-top: 2px solid var(--primary-color);
}

.copyright {
  padding: 0 0 0 25px;
  font-size: 18px;
  letter-spacing: 0.8px;
  line-height: 1.5;
  color: var(--secondary-color);
}

.link-social {
  display: flex;
  gap: 20px;
  align-items: center;
  padding: 15px 0 0 25px;
}

.link-social-item {
  color: var(--secondary-color);
}

.link-social-item:hover {
  color: var(--primary-color-text);
}

.footer-right {
  display: flex;
  gap: 100px;
  padding: 40px 0 0;
}

.footer-column {
  display: flex;
  flex-direction: column;
  gap: 20px;
}

.footer-column-title {
  font-family: var(--primary-font);
  font-size: 28px;
  text-transform: capitalize;
}

.footer-column-items {
  display: flex;
  flex-direction: column;
  gap: 15px 0;
}

.footer-column-link {
  color: var(--secondary-color);
  transition: all 0.3s ease-in-out;
}

.footer-column-link:hover {
  color: var(--primary-color-text);
}

/* Media Query for Laptop Devices */
@media only screen and (max-width: 1024px) {
  .footer-right {
    gap: 50px;
    padding: 25px 0 0;
  }

  .footer-column {
    gap: 20px;
  }
}

/* Media Query for Tablets and Smaller Screens */
@media only screen and (max-width: 768px) {
  .footer-column-title {
    font-size: 20px;
  }

  .footer-column-items {
    gap: 10px 0;
  }

  .footer-right {
    gap: 35px;
    padding: 25px 0 0;
  }

  .copyright {
    font-size: 16px;
    line-height: 1.5;
  }
}

/* Media Query for Mobile Devices */
@media only screen and (max-width: 480px) {
  .site-footer {
    flex-direction: column-reverse;
  }

  .footer-left {
    display: grid;
    grid-template-columns: 1fr 1fr;
  }

  .copyright {
    grid-column: 1;
    grid-row: 1;
    padding: 20px 0;
  }

  .link-social {
    padding: 0;
  }
}
