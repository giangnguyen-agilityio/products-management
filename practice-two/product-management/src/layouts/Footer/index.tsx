// Libraries
import React from 'react'
import { Flex, Text, Image, Box, Container } from '@chakra-ui/react'
import { Link } from 'react-router-dom'

// Components
import Logo from '@components/common/Logo'

// Assets
import logoWebsite from '@assets/images/logo_website.webp'

// Constants
import { aboutUsLinks, footerLinks, helpLinks, socialLinks } from '@constants'

// Types
import { FooterButtonProps, FooterLinkProps, FooterColumnProps } from '@types'

const FooterButton: React.FC<FooterButtonProps> = ({ icon, href }) => (
  <Link to={href}>
    <Box
      marginRight={1}
      opacity={0.5}
      _hover={{
        opacity: '1',
      }}
    >
      <Image src={icon} alt="The social icon" width={8} height={8} />
    </Box>
  </Link>
)

const FooterLink: React.FC<FooterLinkProps> = ({ href, label }) => (
  <Flex as="li" my={1}>
    <Link to={href}>
      <Text
        as="span"
        size="secondary"
        className="footer-link"
        variant="fontPrimaryMedium"
        color="textSecondary"
        _hover={{
          opacity: '0.5',
        }}
      >
        {label}
      </Text>
    </Link>
  </Flex>
)

const FooterColumn: React.FC<FooterColumnProps> = ({ title, links }) => (
  <Flex as="li" className="footer-column" flexDirection="column">
    <Text
      as="span"
      size="primary"
      marginBottom={2}
      color="textSecondary"
      textTransform="uppercase"
      variant="fontPrimaryBold"
    >
      {title}
    </Text>
    <Flex as="ul" className="footer-column-items" direction="column">
      {links.map((link, index) => (
        <FooterLink key={index} href={link.href} label={link.label} />
      ))}
    </Flex>
  </Flex>
)

const Footer: React.FC = () => {
  return (
    <Container as="footer" className="footer">
      <Flex
        flexDirection={{ base: 'column-reverse', lg: 'row' }}
        justifyContent="space-between"
        alignItems="flex-start"
        gap={{ base: 2, sm: 8 }}
      >
        {/* Footer left */}
        <Flex as="ul" className="footer-left" flexDirection="column">
          <Flex as="li">
            <Logo imageSrc={logoWebsite} />
          </Flex>
          <Flex as="li">
            <Text
              as="span"
              pt={2}
              pl={4}
              color="textSecondary"
              size="secondary"
            >
              Copyright &copy; 2023, Giang Nguyen
              <br />
              All rights reserved.
            </Text>
          </Flex>
          {/* Social media links */}
          <Flex as="li" className="link-social" pt={1} pl={2.5}>
            {socialLinks.map((link, index) => (
              <FooterButton key={index} icon={link.icon} href={link.href} />
            ))}
          </Flex>
        </Flex>

        {/* Footer right */}
        <Flex
          as="ul"
          className="footer-right"
          marginLeft={{ base: 4, xl: 0 }}
          gap={{ base: 8, xl: 16 }}
        >
          {/* Column 1 */}
          <FooterColumn title="Discover" links={footerLinks} />

          {/* Column 2 */}
          <FooterColumn title="About Us" links={aboutUsLinks} />

          {/* Column 3 */}
          <FooterColumn title="Help" links={helpLinks} />
        </Flex>
      </Flex>
    </Container>
  )
}

export default Footer
