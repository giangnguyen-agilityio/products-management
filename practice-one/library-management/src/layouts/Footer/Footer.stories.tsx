import {type StoryFn, type Meta} from '@storybook/react'
import {BrowserRouter} from 'react-router-dom'
import Footer from './index'
import React from 'react'

export default {
  title: 'Example/Footer',
  component: Footer,
} as Meta<typeof Footer>

const Template: StoryFn<typeof Footer> = () => (
  <BrowserRouter>
    <Footer />
  </BrowserRouter>
)
export const Default = Template.bind({})
