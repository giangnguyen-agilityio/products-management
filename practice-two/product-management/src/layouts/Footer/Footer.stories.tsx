import type { Meta, StoryObj } from '@storybook/react'
import Footer from './index'

const meta = {
  title: 'Practice Two/Footer Component',
  component: Footer,
  decorators: [
    Story => (
      <div
        style={{
          backgroundColor: '#532EE7',
          color: '#FFF',
          padding: '20px',
        }}
      >
        <Story />
      </div>
    ),
  ],
  parameters: {
    layout: 'top',
  },
  tags: ['autodocs'],
} satisfies Meta<typeof Footer>

export default meta
type Story = StoryObj<typeof meta>

export const Primary: Story = {}
