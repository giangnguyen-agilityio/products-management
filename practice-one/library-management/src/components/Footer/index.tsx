import Logo from '../Logo/index'
import Typography from '../Typography/index'
import logo from '../../assets/images/logo-website-footer.png'
import { BsTwitter, BsInstagram } from 'react-icons/bs'
import { FaFacebookSquare } from 'react-icons/fa'
import './footer.css'

const Footer = () => {
  return (
    <footer className="site-footer">
      <div className="footer-left">
        {/* The website logo */}
        <Logo
          imageSrc={logo}
          altText={'This is the logo website'}
          widthSize={150}
          heightSize={70}
        />
        <Typography variant="p" className="copyright">
          Copyright &copy; 2023, Giang Nguyen
          <br />
          All rights reserved.
        </Typography>
        <ul className="link-social">
          <li>
            <a href="javascript:void(0)" className="link-social-item">
              <BsTwitter size={30} />
            </a>
          </li>
          <li>
            <a href="javascript:void(0)" className="link-social-item">
              <FaFacebookSquare size={30} />
            </a>
          </li>
          <li>
            <a href="javascript:void(0)" className="link-social-item">
              <BsInstagram size={30} />
            </a>
          </li>
        </ul>
      </div>
      <div className="footer-right">
        <div className="footer-column">
          <Typography variant="p" className="footer-column-title">
            discover
          </Typography>
          <ul className="footer-column-items">
            <li>
              <a href="javascript:void(0)" className="footer-column-link">
                Home
              </a>
            </li>
            <li>
              <a href="javascript:void(0)" className="footer-column-link">
                Books
              </a>
            </li>
            <li>
              <a href="javascript:void(0)" className="footer-column-link">
                Authors
              </a>
            </li>
            <li>
              <a href="javascript:void(0)" className="footer-column-link">
                Member
              </a>
            </li>
            <li>
              <a href="javascript:void(0)" className="footer-column-link">
                Hire requests
              </a>
            </li>
          </ul>
        </div>
        <div className="footer-column">
          <Typography variant="p" className="footer-column-title">
            about us
          </Typography>
          <ul className="footer-column-items">
            <li>
              <a href="javascript:void(0)" className="footer-column-link">
                Privacy Policy
              </a>
            </li>
            <li>
              <a href="javascript:void(0)" className="footer-column-link">
                Service Terms
              </a>
            </li>
            <li>
              <a href="javascript:void(0)" className="footer-column-link">
                Contact
              </a>
            </li>
          </ul>
        </div>
        <div className="footer-column">
          <Typography variant="p" className="footer-column-title">
            help
          </Typography>
          <ul className="footer-column-items">
            <li>
              <a href="javascript:void(0)" className="footer-column-link">
                Help Center
              </a>
            </li>
            <li>
              <a href="javascript:void(0)" className="footer-column-link">
                Report a Problem
              </a>
            </li>
            <li>
              <a href="javascript:void(0)" className="footer-column-link">
                Contact Us
              </a>
            </li>
          </ul>
        </div>
      </div>
    </footer>
  )
}

export default Footer
