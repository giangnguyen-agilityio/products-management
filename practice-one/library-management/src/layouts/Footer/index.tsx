import React from 'react'
import Logo from '@components/commons/Logo'
import Typography from '@components/commons/Typography'
import logo from '@assets/images/logo-website-footer.png'
import {BsTwitter, BsInstagram} from 'react-icons/bs'
import {FaFacebookSquare} from 'react-icons/fa'
import './footer.css'

const Footer: React.FC = () => {
  return (
    <footer className="site-footer">
      <div className="site-footer-content">
        {/* The footer left */}
        <div className="footer-left">
          {/* The website logo */}
          <Logo
            imageSrc={logo}
            altText="This is the logo website"
            widthSize={150}
            heightSize={70}
          />
          {/* The website copyright */}
          <Typography variant="p" className="copyright">
            Copyright &copy; 2023, Giang Nguyen
            <br />
            All rights reserved.
          </Typography>
          {/* The website link social */}
          <ul className="link-social">
            <li>
              <a href="#" className="link-social-item">
                <BsTwitter size={30} />
              </a>
            </li>
            <li>
              <a href="#" className="link-social-item">
                <FaFacebookSquare size={30} />
              </a>
            </li>
            <li>
              <a href="#" className="link-social-item">
                <BsInstagram size={30} />
              </a>
            </li>
          </ul>
        </div>
        {/* The footer right */}
        <div className="footer-right">
          <div className="footer-column">
            <Typography variant="p" className="footer-column-title">
              discover
            </Typography>
            <ul className="footer-column-items">
              <li>
                <a href="#" className="footer-column-link">
                  Home
                </a>
              </li>
              <li>
                <a href="#" className="footer-column-link">
                  Books
                </a>
              </li>
              <li>
                <a href="#" className="footer-column-link">
                  Authors
                </a>
              </li>
              <li>
                <a href="#" className="footer-column-link">
                  Member
                </a>
              </li>
              <li>
                <a href="#" className="footer-column-link">
                  Hire requests
                </a>
              </li>
            </ul>
          </div>
          <div className="footer-column">
            <Typography variant="p" className="footer-column-title">
              about us
            </Typography>
            <ul className="footer-column-items">
              <li>
                <a href="#" className="footer-column-link">
                  Privacy Policy
                </a>
              </li>
              <li>
                <a href="#" className="footer-column-link">
                  Service Terms
                </a>
              </li>
              <li>
                <a href="#" className="footer-column-link">
                  Contact
                </a>
              </li>
            </ul>
          </div>
          <div className="footer-column">
            <Typography variant="p" className="footer-column-title">
              help
            </Typography>
            <ul className="footer-column-items">
              <li>
                <a href="#" className="footer-column-link">
                  Help Center
                </a>
              </li>
              <li>
                <a href="#" className="footer-column-link">
                  Report a Problem
                </a>
              </li>
              <li>
                <a href="#" className="footer-column-link">
                  Contact Us
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </footer>
  )
}

export default Footer
