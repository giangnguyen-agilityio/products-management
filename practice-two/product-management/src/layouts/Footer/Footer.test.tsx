import { render } from '@testing-library/react'
import { BrowserRouter as Router } from 'react-router-dom'
import { ChakraProvider } from '@chakra-ui/react'
import Footer from '.'

describe('Footer component', () => {
  it('renders without crashing', () => {
    const { container } = render(
      <ChakraProvider>
        <Router>
          <Footer />
        </Router>
      </ChakraProvider>
    )
    expect(container).toMatchSnapshot()
  })
})
