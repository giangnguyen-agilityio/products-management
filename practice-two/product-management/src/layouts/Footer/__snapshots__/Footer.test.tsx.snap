// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component renders without crashing 1`] = `
<div>
  <footer
    class="chakra-container footer css-vhc7ah"
  >
    <div
      class="css-hv056j"
    >
      <ul
        class="footer-left css-j7qwjs"
      >
        <li
          class="css-k008qs"
        >
          <a
            href="/"
          >
            <div
              class="logo css-759u60"
            />
          </a>
        </li>
        <li
          class="css-k008qs"
        >
          <span
            class="chakra-text css-lsakg3"
          >
            Copyright Â© 2023, Giang Nguyen
            <br />
            All rights reserved.
          </span>
        </li>
        <li
          class="link-social css-1ov7iep"
        >
          <a
            href="/"
          >
            <div
              class="css-1wojifn"
            >
              <img
                alt="The social icon"
                class="chakra-image css-v67zxb"
                src=""
              />
            </div>
          </a>
          <a
            href="/"
          >
            <div
              class="css-1wojifn"
            >
              <img
                alt="The social icon"
                class="chakra-image css-v67zxb"
                src=""
              />
            </div>
          </a>
          <a
            href="/"
          >
            <div
              class="css-1wojifn"
            >
              <img
                alt="The social icon"
                class="chakra-image css-v67zxb"
                src=""
              />
            </div>
          </a>
        </li>
      </ul>
      <ul
        class="footer-right css-1jnspk1"
      >
        <li
          class="footer-column css-j7qwjs"
        >
          <span
            class="chakra-text css-1c6t563"
          >
            Discover
          </span>
          <ul
            class="footer-column-items css-j7qwjs"
          >
            <li
              class="css-1l0pjoz"
            >
              <a
                href="/"
              >
                <span
                  class="chakra-text footer-link css-14cj6ry"
                >
                  Home
                </span>
              </a>
            </li>
            <li
              class="css-1l0pjoz"
            >
              <a
                href="/"
              >
                <span
                  class="chakra-text footer-link css-14cj6ry"
                >
                  Products
                </span>
              </a>
            </li>
          </ul>
        </li>
        <li
          class="footer-column css-j7qwjs"
        >
          <span
            class="chakra-text css-1c6t563"
          >
            About Us
          </span>
          <ul
            class="footer-column-items css-j7qwjs"
          >
            <li
              class="css-1l0pjoz"
            >
              <a
                href="/"
              >
                <span
                  class="chakra-text footer-link css-14cj6ry"
                >
                  Privacy Policy
                </span>
              </a>
            </li>
            <li
              class="css-1l0pjoz"
            >
              <a
                href="/"
              >
                <span
                  class="chakra-text footer-link css-14cj6ry"
                >
                  Service Terms
                </span>
              </a>
            </li>
            <li
              class="css-1l0pjoz"
            >
              <a
                href="/"
              >
                <span
                  class="chakra-text footer-link css-14cj6ry"
                >
                  Contact
                </span>
              </a>
            </li>
          </ul>
        </li>
        <li
          class="footer-column css-j7qwjs"
        >
          <span
            class="chakra-text css-1c6t563"
          >
            Help
          </span>
          <ul
            class="footer-column-items css-j7qwjs"
          >
            <li
              class="css-1l0pjoz"
            >
              <a
                href="/"
              >
                <span
                  class="chakra-text footer-link css-14cj6ry"
                >
                  Help Center
                </span>
              </a>
            </li>
            <li
              class="css-1l0pjoz"
            >
              <a
                href="/"
              >
                <span
                  class="chakra-text footer-link css-14cj6ry"
                >
                  Report a Problem
                </span>
              </a>
            </li>
          </ul>
        </li>
      </ul>
    </div>
  </footer>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
